subsys/usb/device_next/class/usbd_cdc_ncm.c

```diff
@@ -0,0 +1,1282 @@
+		return 0;
+	}
+
+	LOG_WRN("usbd_cdc_ncm_control_to_device - bmRequestType 0x%02x bRequest 0x%02x unsupported",
+			setup->bmRequestType, setup->bRequest);
+	return -ENOTSUP;
+}   /* usbd_cdc_ncm_control_to_device */
+
+
+
+static int usbd_cdc_ncm_control_to_host(struct usbd_class_data *const c_data,
+		const struct usb_setup_packet *const setup,
+		struct net_buf *const buf)
+/**
+ * USB control request handler to host
+ */
+{
+	LOG_DBG("---------------------------------------------------"
+		" req_type 0x%x req 0x%x  buf %p",
+		setup->bmRequestType, setup->bRequest, buf);
+
+	switch (setup->RequestType.type) {
+	case USB_REQTYPE_TYPE_STANDARD:
+		LOG_DBG("  USB_REQTYPE_TYPE_STANDARD: %d %d %d %d", setup->bRequest,
+				setup->wValue, setup->wIndex, setup->wLength);
+		break;
+
+	case USB_REQTYPE_TYPE_CLASS:
+		LOG_DBG("  USB_REQTYPE_TYPE_CLASS: %d %d %d %d", setup->bRequest,
+				setup->wLength, setup->wIndex, setup->wValue);
+
+		if (setup->bRequest == NCM_GET_NTB_PARAMETERS) {
+			LOG_DBG("    NCM_GET_NTB_PARAMETERS");
+			net_buf_add_mem(buf, &ntb_parameters, sizeof(ntb_parameters));
+			return 0;
+		} else if (setup->bRequest == NCM_SET_ETHERNET_PACKET_FILTER) {
+			LOG_WRN("    NCM_SET_ETHERNET_PACKET_FILTER (not supported)");
+			return -ENOTSUP;
+		} else if (setup->bRequest == NCM_GET_NTB_INPUT_SIZE) {
+			LOG_ERR("    NCM_GET_NTB_INPUT_SIZE (not supported, but required)");
+			return -ENOTSUP;
+		} else if (setup->bRequest == NCM_SET_NTB_INPUT_SIZE) {
+			LOG_ERR("    NCM_SET_NTB_INPUT_SIZE (not supported, but required)");
+			return -ENOTSUP;
+		}
+		LOG_WRN("    not supported: %d", setup->bRequest);
+		return -ENOTSUP;
+
+	default:
+		/* unsupported request */
+		return -ENOTSUP;
+	}
+
+	return -ENOTSUP;
+}   /* usbd_cdc_ncm_control_to_host */
+
+
+
+static int usbd_cdc_ncm_init(struct usbd_class_data *const c_data)
+/**
+ * Initialization of the class implementation
+ */
+{
+	struct usbd_contex *uds_ctx = usbd_class_get_ctx(c_data);
+	const struct device *dev = usbd_class_get_private(c_data);
+	struct cdc_ncm_eth_data *const data = dev->data;
+	struct usbd_cdc_ncm_desc *desc = data->desc;
+	const uint8_t if_num = desc->if0.bInterfaceNumber;
+
+	/* Update relevant b*Interface fields */
+	desc->iad.bFirstInterface = if_num;
+	desc->if0_union.bControlInterface = if_num;
+	desc->if0_union.bSubordinateInterface0 = if_num + 1;
+	LOG_DBG("CDC NCM class initialized %d", if_num);
+
+	if (usbd_add_descriptor(uds_ctx, data->mac_desc_data)) {
+		LOG_ERR("Failed to add iMACAddress string descriptor");
+	} else {
+		desc->if0_ncm.iMACAddress = data->mac_desc_data->idx;
+	}
+
+	return 0;
+}   /* usbd_cdc_ncm_init */
+
+
+
+static void usbd_cdc_ncm_shutdown(struct usbd_class_data *const c_data)
+/**
+ * Shutdown of the class implementation
+ */
+{
+	const struct device *dev = usbd_class_get_private(c_data);
+	struct cdc_ncm_eth_data *const data = dev->data;
+	struct usbd_cdc_ncm_desc *desc = data->desc;
+
+	LOG_DBG("");
```

Comment:
Yes of course.  The driver currently lacks some relevant features and it is in experimental state.  So I decided to leave those traces in, except anyone wishes otherwise.